### DISCUSSION TITLE
Can you mount with a specific group?

### DISCUSSION THREAD

**Discussion by danielkrajnik** [trapexit/mergerfs#1279] [Q&A] [answered]
I'm trying to mount mergerfs with a specific group, but I can't seem to find it documented anywhere. Is this possible at all?

```sh
/usr/bin/mergerfs \
  -f \
  -o cache.files=partial \
  -o dropcacheonclose=true \
  /mnt/mount1:/mnt/mount2=NC \
  /mnt/merger
```

I've tried adding `-o gid=1001` or `-o group=new-group`, but nothing works (it always takes the group of the current user).

**Comment by trapexit** [comment]
What do you mean by "mount with a specific group"?

**Reply by danielkrajnik** [reply]
Thank you for the quick reply. I'm trying to make the `/mnt/merger` folder's group permissions to be different than the user, e.g. `ls -l /mnt` :
```
drwxrwx--- - user new-group 26 Jun 17:56  merger/
```
Even if you change permissions for that folder before mounting mergerfs it still overrides them.

**Reply by trapexit** [reply, marked as answer]
Can you explain the usecase? Why not just change the ownership of the paths?

As with most libfuse apps mergerfs has -o gid=x and uid=y to override values. If that doesn't work for your purposes I'm not sure what exactly you're looking for.

**Reply by danielkrajnik** [reply]
> As with most libfuse apps mergerfs has -o gid=x and uid=y to override values

Thanks, that helped. I forgot that you need to also specify uid (I was only passing gid). This works:

```
  -o uid=1000 \
  -o gid=1001 \
  -o umask=007 \
```

Sadly it didn't solve my problem, but that's unrelated to mergerfs. I'm trying to pass the filesystem to a virtual machine via virtiofs daemon, which runs under a different user (libvirt-qemu). Normally sharing filesystem via samba doesn't require different permissions, because it runs under the user. This however keeps giving this cryptic error: `the virtiofs export directory '/mnt/merger' does not exist` . I thought that was caused by permissions, but it's still the same even after changing them. Oh well...
